ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml

```diff
@@ -586,6 +585,55 @@ let with_polly_wait kind fd f =
       in
       f wait fd
 
+module Timeout = struct
+  type t = Mtime.Span.t
+
+  let of_span s : t = s
+
+  let of_string str =
+    let seconds = Float.of_string str in
+    match Mtime.Span.of_float_ns (seconds *. 1e9 |> Float.round) with
+    | Some span ->
+        span
+    | None ->
+        (* e.g. negative, or >=2^53. *)
+        invalid_arg (Printf.sprintf "Span out of range: %gs" seconds)
+
+  (* nanosecond conversion doesn't roundtrip, so use only microseconds, which does *)
+  let to_string t = Mtime.Span.to_float_ns t *. 1e-9 |> Printf.sprintf "%.6f"
```

Comment:
Is the comment above correct?

Reply:
Yes, I tried using nanoseconds, but multiplying and dividing by `1e-9` results in something different to the original. It could probably be fixed by careful use of rounding, although some of this happens inside Mtime.